python/S3_OBJECT_LOCK_ENABLED/S3_OBJECT_LOCK_ENABLED.py

```diff
@@ -0,0 +1,184 @@
+  Scenario: 8
+    Given: S3 bucket is object lock enabled with mode and days
+          (actual bucket days are less than the parameter days)
+     Then: Return NON_COMPLIANT
+  Scenario: 9
+    Given: S3 bucket is object lock enabled with mode and years
+          (actual bucket years are less than the parameter years)
+     Then: Return NON_COMPLIANT
+  Scenario: 10
+    Given: S3 bucket is object lock enabled with mode, days and yeaas
+          (actual bucket days(days + (years*365)) are less than the parameter days)
+     Then: Return NON_COMPLIANT
+'''
+
+import json
+
+from rdklib import Evaluator, Evaluation, ConfigRule, ComplianceType, InvalidParametersError
+
+RESOURCE_TYPE = 'AWS::S3::Bucket'
+
+class S3_OBJECT_LOCK_ENABLED(ConfigRule):
+    def evaluate_periodic(self, event, client_factory, valid_rule_parameters):
+        s3_client = client_factory.build_client("s3") # AWS S3 Client
+        config_client = client_factory.build_client('config') # AWS Config Service Client
+        evaluations = []
+        valid_rule_parameters = self.evaluate_parameters(valid_rule_parameters)
+        for s3_bucket_name in get_buckets(config_client):
+
+            isexception = False
+            try:
+                response = s3_client.get_object_lock_configuration(Bucket=s3_bucket_name)
+            except:
+                isexception = True
+            if isexception or 'ObjectLockConfiguration' not in response:
+                evaluations.append(
+                    Evaluation(ComplianceType.NON_COMPLIANT,
+                               s3_bucket_name, RESOURCE_TYPE,
+                               "No ObjectLockConfiguration exist for this bucket.")
+                )
+            else:
+                days = 0
+                years = 0
+                mode = None
+                #If 'Rule' exists in the response, then only derive 'mode'
+                if response.get('ObjectLockConfiguration').get('Rule', 0) != 0:
+                    mode = response['ObjectLockConfiguration']['Rule']['DefaultRetention'].get('Mode', None)
+
+                #If 'Days' exists in the response, then only derive 'Days' otherwise make it 0
+                if response.get('ObjectLockConfiguration').get('Rule').get('DefaultRetention').get('Days', 0) > 0:
+                    days = response['ObjectLockConfiguration']['Rule']['DefaultRetention']['Days']
+
+                #If 'Years' exists in the response, then only derive 'Years' otherwise make it 0
+                if response.get('ObjectLockConfiguration').get('Rule').get('DefaultRetention').get('Years', 0) > 0:
+                    years = response['ObjectLockConfiguration']['Rule']['DefaultRetention']['Years']
+
+                days_in_param = valid_rule_parameters.get('Days', 0)
+                years_in_param = valid_rule_parameters.get('Years', 0)
+                #Calculate days from years
+                days_in_param = (years_in_param * 365) + days_in_param
```

Comment:
No. years_in_param and day_in_param should be exclusive. Maybe we should reduce to only the day parameters. then do the math if "Years" is configured in the ObjectRetention configuration. It'd be more customer friendly. Thoughts?